pow-migration/src/state/mod.rs

```diff
@@ -392,27 +397,38 @@ pub async fn get_stakers(
                                 log::info!(staker_address=txn.from_address, validator_address=%address, "Found pre-stake transaction for validator");
                                 if let Ok(staker_address) = Address::from_str(&txn.from_address) {
                                     let stake = Coin::from_u64_unchecked(txn.value);
-                                    if stake >= Coin::from_u64_unchecked(Policy::MINIMUM_STAKE) {
-                                        validator.total_stake += stake;
-                                        stakers
-                                            .entry(staker_address.clone())
-                                            .and_modify(|staker| {
-                                                // If we have an entry for this staker, treat this as a switch to another
-                                                // validator or an increase of the stake (if the validator isn't changed)
-                                                staker.delegation =
-                                                    validator.validator.validator_address.clone();
-                                                staker.balance += stake;
-                                            })
-                                            .or_insert(GenesisStaker {
-                                                staker_address,
-                                                balance: stake,
-                                                delegation: validator
-                                                    .validator
-                                                    .validator_address
-                                                    .clone(),
-                                                inactive_balance: Coin::ZERO,
-                                                inactive_from: None,
-                                            });
+                                    match stakers.entry(staker_address.clone()) {
+                                        Entry::Occupied(mut entry) => {
+                                            // If we have an entry for this staker, treat this as a switch to another
+                                            // validator or an increase of the stake (if the validator isn't changed)
+                                            let staker = entry.get_mut();
+                                            staker.delegation =
+                                                validator.validator.validator_address.clone();
+                                            staker.balance += stake;
+
+                                            // Update the validator total stake
+                                            validator.total_stake += stake;
```

Comment:
Shouldn't we also decrease the `total_stake` of the old validator if we switching between validators?

Reply:
Yup, you are right. Added logic to address this